pkg/apis/core/types_shootproviderstate.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShootProviderState holds the provider status information of extensions of a Shoot cluster
+type ShootProviderState struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	// Designed to have an owner reference to the associated Shoot resource
+	metav1.ObjectMeta
+	// Specification of the ShootProviderState.
+	Spec ShootProviderStateSpec
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShootProviderStateList is a list of ShootProviderState objects.
+type ShootProviderStateList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of ShootProviderStates.
+	Items []ShootProviderState
+}
+
+// ShootProviderStateSpec is the specification of the ShootProviderState.
+type ShootProviderStateSpec struct {
+	// ProviderStatus holds a list containing the information of the providerStatus field of Extension resources
+	// in the Seed cluster
+	ProviderStatus []ExtensionProviderState
+}
+
+// ExtensionProviderState contains the kind, the type and the last observed provider status
+// of a Gardener extension resource from a Shoot's namespace in the Seed cluster.
+type ExtensionProviderState struct {
+	// Kind of the extension resource
+	Kind string
+	// Type of the extension resource
+	Type *string
+	// Status is the provider status of the extension resource
+	Status *runtime.RawExtension
```

Comment:
```suggestion
	Status runtime.Object
```

Let's follow the recommendation from the doc string of `RawExtension` (see [here](https://github.com/kubernetes/apimachinery/blob/d23dd757f6df1aa5f84b372fefbe4249d3e59943/pkg/runtime/types.go#L49)).